src/main/scala/beam/utils/plan/PlansBuilder.scala

```diff
@@ -123,7 +122,7 @@ object PlansBuilder {
     val permissibleModes = modeAllocator
       .getPermissibleModes(person.getSelectedPlan)
       .asScala
-    AvailableModeUtils.setAvailableModesForPerson(person, newPop, permissibleModes.toSeq)
+    setAvailableModesForPerson(person, newPop, permissibleModes.toSeq)
```

Comment:
Actually, this one I'm not a fan of. I prefer taking it to the final object and then leaving it there when it comes to statics. Otherwise it makes it just a bit harder to trace where the method is defined - especially for new devs

Reply:
This PR is a pre-step to fulfil the the issue https://github.com/LBNL-UCB-STI/beam/issues/1853

Having an "static" object to apply mutations to an object that should be immutable in the first place looks like spreading the problem around the system.

The final goal here is moving the responsibility yo the loader (`BeamScenarioLoader`). There is no reason to load scenario partially and in other steps fix the inconsistent scenario. 

To sum up: ScenarioLoader will be responsible to load persons (like is already is) and its attributes (like is partially is). Ideally the scenario will become immutable and once is loaded cannot change anymore

Reply:
in other words: mutating a state passed as parameter is a bad practice.
```
setAvailableModesForPerson(person, newPop, permissibleModes.toSeq)
```
will be replaced with
```
person.availableModes = calculateAvailableModes(newPop, permissibleModes.toSeq)
```
when building the scenario

Reply:
done